com.unity.ml-agents.extensions/Runtime/Sensors/BoxOverlapChecker.cs

```diff
@@ -0,0 +1,247 @@
+            {
+                numFound = Physics.OverlapBoxNonAlloc(cellCenter, halfCellScale, m_ColliderBuffer, rotation, m_ColliderMask);
+                if (numFound == m_ColliderBuffer.Length && m_ColliderBuffer.Length < m_MaxColliderBufferSize)
+                {
+                    m_ColliderBuffer = new Collider[Math.Min(m_MaxColliderBufferSize, m_ColliderBuffer.Length * 2)];
+                    m_InitialColliderBufferSize = m_ColliderBuffer.Length;
+                }
+                else
+                {
+                    break;
+                }
+            }
+            return numFound;
+        }
+
+        /// <summary>
+        /// Perceive the latest grid status. Call OverlapBoxNonAlloc once to detect colliders.
+        /// Then parse the collider arrays according to all available gridSensor delegates.
+        /// </summary>
+        internal void Update()
+        {
+            for (var cellIndex = 0; cellIndex < m_NumCells; cellIndex++)
+            {
+                var cellCenter = GetCellGlobalPosition(cellIndex);
+                var numFound = BufferResizingOverlapBoxNonAlloc(cellCenter, m_HalfCellScale, GetGridRotation());
+
+                if (GridOverlapDetectedAll != null)
+                {
+                    ParseCollidersAll(m_ColliderBuffer, numFound, cellIndex, cellCenter, GridOverlapDetectedAll);
+                }
+                if (GridOverlapDetectedClosest != null)
+                {
+                    ParseCollidersClosest(m_ColliderBuffer, numFound, cellIndex, cellCenter, GridOverlapDetectedClosest);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Same as Update(), but only load data for debug gizmo.
+        /// </summary>
+        internal void UpdateGizmo()
+        {
+            for (var cellIndex = 0; cellIndex < m_NumCells; cellIndex++)
+            {
+                var cellCenter = GetCellGlobalPosition(cellIndex);
+                var numFound = BufferResizingOverlapBoxNonAlloc(cellCenter, m_HalfCellScale, GetGridRotation());
+
+                ParseCollidersClosest(m_ColliderBuffer, numFound, cellIndex, cellCenter, GridOverlapDetectedDebug);
+            }
+        }
+
+        /// <summary>
+        /// Parses the array of colliders found within a cell. Finds the closest gameobject to the agent root reference within the cell
+        /// </summary>
+        void ParseCollidersClosest(Collider[] foundColliders, int numFound, int cellIndex, Vector3 cellCenter, Action<GameObject, int> detectedAction)
+        {
+            GameObject closestColliderGo = null;
+            var minDistanceSquared = float.MaxValue;
+
+            for (var i = 0; i < numFound; i++)
+            {
+                var currentColliderGo = foundColliders[i].gameObject;
+
+                // Continue if the current collider go is the root reference
+                if (ReferenceEquals(currentColliderGo, m_RootReference))
+                    continue;
```

Comment:
nit:
```suggestion
                {
                    continue;
                }
```
(this is probably from the original code)